src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -0,0 +1,512 @@
+                    onData(data);
+                    break;
+                case EndFW.TYPE_ID:
+                    final EndFW end = endRO.wrap(buffer, index, index + length);
+                    onEnd(end);
+                    break;
+                case AbortFW.TYPE_ID:
+                    final AbortFW abort = abortRO.wrap(buffer, index, index + length);
+                    onAbort(abort);
+                    break;
+                case WindowFW.TYPE_ID:
+                    final WindowFW window = windowRO.wrap(buffer, index, index + length);
+                    onWindow(window);
+                    break;
+                case ResetFW.TYPE_ID:
+                    final ResetFW reset = resetRO.wrap(buffer, index, index + length);
+                    onReset(reset);
+                    break;
+                case SignalFW.TYPE_ID:
+                    final SignalFW signal = signalRO.wrap(buffer, index, index + length);
+                    onSignal(signal);
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        private void onBegin(
+            BeginFW begin)
+        {
+            doBegin(supplyTraceId.getAsLong());
+        }
+
+        private void onData(
+            DataFW data)
+        {
+            initialBudget -= data.length() + data.padding();
+
+            if (initialBudget < 0)
+            {
+                doReset(supplyTraceId.getAsLong());
+            }
+            else
+            {
+                decodeTraceId = data.trace();
+                final OctetsFW payload = data.payload();
+                final DirectBuffer buffer = payload.buffer();
+
+                int offset = payload.offset();
+                int length = payload.sizeof();
+
+                while (length > 0)
+                {
+                    int consumed = decodeState.decode(buffer, offset, length);
+                    offset += consumed;
+                    length -= consumed;
+                }
+            }
+        }
+
+        private void onEnd(
+            EndFW end)
+        {
+            // TODO
+        }
+
+        private void onAbort(
+            AbortFW abort)
+        {
+            // TODO
+        }
+
+        private void onWindow(
+            WindowFW window)
+        {
+            final int replyCredit = window.credit();
+
+            replyBudget += replyCredit;
+            replyPadding += window.padding();
+
+            final int initialCredit = bufferPool.slotCapacity() - initialBudget;
+            doWindow(supplyTraceId.getAsLong(), initialCredit);
+        }
+
+        private void onReset(
+            ResetFW reset)
+        {
+            // TODO
+        }
+
+        private void onSignal(
+            SignalFW signal)
+        {
+            // TODO
+        }
+
+        private void onAmqpHeader(AmqpHeaderFW header)
+        {
+            doAmqpHeader(header.buffer(), header.offset(), header.limit());
```

Comment:
We need some validation here to ensure we are speaking a consistent version of `AMQP 1.0`.
If the initial bytes do not represent a valid header exchange, then the connection should be rejected with appropriate error code. Likely we need to add this as a negative spec test in `nukleus-amqp.spec`.